src/lib.rs

```diff
@@ -555,16 +553,23 @@ impl <'a, T> FixedVec<'a, T> where T: 'a + Copy {
     /// vec.push_all(&[1, 2, 3]).unwrap();
     /// {
     ///     let mut iter = vec.iter();
-    ///     assert_eq!(iter.next(), Some(1));
-    ///     assert_eq!(iter.next(), Some(2));
-    ///     assert_eq!(iter.next(), Some(3));
+    ///     assert_eq!(iter.next(), Some(&1));
+    ///     assert_eq!(iter.next(), Some(&2));
+    ///     assert_eq!(iter.next(), Some(&3));
     ///     assert_eq!(iter.next(), None);
     /// }
     /// # }
     /// ```
     #[inline]
     pub fn iter(&'a self) -> Iter<'a, T> {
```

Comment:
The `'a` lifetimes are no longer needed here since the slice now manages the iterator's lifetime.